src/gt.rs

```diff
@@ -300,58 +293,121 @@ impl Group for Gt {
                     ]),
                 ),
             ),
-        )
-        .0)
+        ))
     }
 
     fn is_identity(&self) -> Choice {
-        Fp12(self.0).ct_eq(&Fp12(Self::identity().0))
+        self.0.ct_eq(&Self::identity().0)
     }
 
     #[must_use]
     fn double(&self) -> Self {
-        Gt(Fp12(self.0).square().0)
+        Gt(self.0.square())
     }
 }
 
 /// Compressed representation of `Fp12`.
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
 #[repr(transparent)]
-pub struct GtCompressed(pub(crate) Fp12Compressed);
+pub struct GtCompressed(pub(crate) Fp6);
 
 impl Gt {
     /// Compress this point. Returns `None` if the element is not in the cyclomtomic subgroup.
     pub fn compress(&self) -> Option<GtCompressed> {
-        Fp12::from(self.0).compress().map(GtCompressed)
+        if !self.is_in_subgroup() {
```

Comment:
Do you think it's necessary to do this check here ? Given the API, is there a way to get Gt element which is not in the group ?